source/System.cpp

```diff
@@ -310,7 +310,29 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
 		else if(key == "object")
 			LoadObject(child, planets);
 		else if(key == "arrival")
-			extraArrivalDistance = child.Value(valueIndex);
+		{
+			if(child.Size() >= 2)
+			{
+				extraHyperArrivalDistance = child.Value(1);
+				extraJumpArrivalDistance = fabs(child.Value(1));
+			}
+			for(const DataNode &grand : child)
+			{
+				if(grand.Size() < 2)
+					child.PrintTrace("Warning: Keywords below arrival always need a value behind the keyword:");
+				else if(grand.Token(0) == "link")
+					extraHyperArrivalDistance = grand.Value(1);
+				else if(grand.Token(0) == "jump")
+					// Negative jump distances work the same as positive jump
+					// distances.
+					// Jump drives use a circle around the target for targeting,
+					// so a value below 0 doesn't have the same meaning as for hyper
+					// drives where negative values would be beyond the target.
```

Comment:
I think it makes more sense to move this comment to the first occurrence of setting this variable, i.e. at line 317.

Reply:
I would remove these comments entirely, and include this clarifying information in the associated header's variable declarations.